doc/mkdocs/docs/features/macros.md

```diff
@@ -12,6 +12,14 @@ This macro overrides `#!cpp catch` calls inside the library. The argument is the
 
 See [Switch off exceptions](../home/exceptions.md#switch-off-exceptions) for an example.
 
+## `JSON_DIAGNOSTICS`
+
+This macro enables extended diagnostics for exception messages. Possible values are `1` to enable or `0` to disable (default).
+
+When enabled, exception messages contain a [JSON Pointer](json_pointer.md) to the JSON value that triggered the exception, see [Extended diagnostic messages](../home/exceptions.md#extended-diagnostic-messages) for an example.
+
+The diagnostics messages can also be controlled with the CMake option `JSON_Diagnostics` (`OFF` by default) which sets `JSON_DIAGNOSTICS` accordingly.
+
```

Comment:
I think this should this mention that it increases the size of every object by one pointer, and makes some operations take a little bit longer, so that people know the cost of enabling it.  Is this pointer available for users to read, or is it completely private?  I don't remember seeing anything that would allow it.

Reply:
Since you already write that in exceptions.md, perhaps just a reference to those docs.

Reply:
Done in 7b7da08fb643c75dd18516ac8835ce2f2d5509a6.